torchvision/models/video/mvit.py

```diff
@@ -177,24 +273,52 @@ def __init__(
                 norm_layer(self.head_dim),
             )
 
+        self.rel_pos_h: Optional[nn.Parameter] = None
```

Comment:
nit: I would suggest to move this whole block to a separate method to make the init less lengthy and more readable.

Reply:
Yes you are right this is quite messy. I'm happy to make changes to make it more readable. Concerning moving this on separate method, here are few concerns:
1. It can be invoked by users after FX and the initialization wont work any more. This is why I moved previously all inits in the constructors.
2. The idiom can in principle lead to weird inheritance issues as the object is not instantiated fully yet. This won't be the problem in this specific case, so it's more of an argument of why I try to avoid the idiom.

What do you think?

Reply:
I agree that it does become problematic in some non standard usage patterns. This was just a nit, so  given your explanation the PR is good to go from my side.

Reply:
Thanks a lot for the flexibility.